src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/Dependencies/IDependencyCollection.cs

```diff
@@ -1,18 +1,49 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System;
 using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
 
 namespace Microsoft.DotNet.UpgradeAssistant.Dependencies
 {
     public interface IDependencyCollection<T> : IEnumerable<T>
     {
+        bool Add(T item, OperationDetails details);
+
+        bool Remove(T item, OperationDetails details);
+
+        [Obsolete("Use the newer Add API", error: false)]
```

Comment:
nit: I'd explain a bit more, such as "This API will be removed sometime after September 1, 2021. Please refactor to use a supported overload"